app/src/main/java/com/codesoom/assignment/domain/UserRepository.java

```diff
@@ -2,14 +2,47 @@
 
 import java.util.Optional;
 
+/**
+ * 유저 저장소.
+ */
 public interface UserRepository {
+    /**
+     * 전달받은 유저를 저장하고 반환합니다.
+     *
+     * @param user 유저
+     * @return 저장된 유저
+     */
     User save(User user);
 
+    /**
+     * 전달받은 email과 일치하는 유저의 존재 여부를 반환합니다.
```

Comment:
`여부`라는 단어는 컨텍스트에 따라 의미가 모호할 수 있습니다.
```suggestion
     * 전달받은 email과 일치하는 유저가 존재한다면 true를, 그렇지 않다면 false를 리턴합니다.
```

Reply:
그렇군요 .. 다만 궁금해지는 것이 말씀해주신 `여부`의 의미가 모호해 지는 경우가 어떤 경우가 있을까요 ?

궁금해서 String 클래스의 javadoc을 보니 아래와 같이 반환타입이 boolean 메소드들의 표현이 조금씩 다른 것을 볼 수 있었는데요
표현이 다른 이유는 스타일 차이라고 보면 될까요 ?

<img width="462" alt="Screen Shot 2021-03-17 at 6 03 20 PM" src="https://user-images.githubusercontent.com/18481400/111442289-89da1580-874b-11eb-9054-2edf07cfc9d2.png">

- (만약 이렇다면 .. ) true를 반환한다.

<img width="461" alt="Screen Shot 2021-03-17 at 6 02 31 PM" src="https://user-images.githubusercontent.com/18481400/111442312-9199ba00-874b-11eb-8e64-c43f33a7d18d.png">

- (만약 이렇다면 .. ) true를 하고 그렇지 않다면 false를 반환한다.

<img width="406" alt="Screen Shot 2021-03-17 at 6 01 55 PM" src="https://user-images.githubusercontent.com/18481400/111442329-978f9b00-874b-11eb-9578-5f2ed1ac0e45.png">

- `Test` 라는 표현은 신선했습니다..

Reply:
표현이 다른 이유는 작성자들이 다르기도 하고(모두 사람입니다 :-) ), 미묘한 차이가 있기 때문이기도 합니다. `여부`를 제가 좋지 않다고 생각하는 이유는 `여부`가 boolean을 의미하는 두 값을 한꺼번에 표현하기 때문이에요.

다음과 같은 코드가 가장 심각한 경우입니다.

```
/* 인증 기간이 지나지 않았는지 여부. */
```
괜찮은 습관을 갖춘 프로그래머라면 이렇게 작성하지 않겠지만, 오랜 세월을 거친 방대한 코드를 유지보수하다보면 숱하게 만납니다. 이런 주석을 보면 주석만으로 판단을 할 수가 없죠. 몇 줄 아래로 내려가서 구현체를 봐야 합니다.

하지만 다음과 같이 작성하면 주석만 보고도 1초 안에 상황 파악이 끝납니다.
```
/* 인증 기간이 지났다면 true, 지나지 않았다면 false를 리턴합니다. */
```

Reply:
개인적으로는 `여부`보다는 `~면 true, 아니면 false`가 더 명확하게 이해되는 것 같아요. 저에게 `여부`와 비슷한 단어로는 `확인`이나 `검증(validate)`이 있어요.

```java
if(check_request(request)){
    do_something();
} else{
    do_something();
}
```

위에 코드가 있고 check_request함수의 주석에 "request가 올바른지 확인합니다"만 주석에 있다면 저는 `true를 반환하면 문제가 없다는걸까? false를 반환하면 문제가 없다는걸까?`라는 생각이 들 것 같아요.

Reply:
피드백 감사합니다 :):) 
생각해보면 `/* 인증 기간이 지나지 않았는지 여부. */` 라는 주석은 고개를 조금만 돌려도 어렵지 않게 볼 수 있는 것 같습니다. 
(물론 그 중에는 제가 작성한 주석도 있다는 게 함정이지만 😂)
`별 문제 없어 보인다는 점` 이라는 말이 정말 무섭게 다가오는 군요.

라스님의 설명 해주신 예제는 더 현실적이어서 이해하는데 많은 도움이 되었습니다. 
여기까지 와서 피드백 해주시다니 정말 감사해요 🙏🙂